Assets/MixedRealityToolkit/Utilities/StandardShader/MeshSmoother.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Component which can be used to automatically generate smoothed normals on a mesh and pack 
+    /// those normals into a UV set. Smoothed normals can be used for a variety of effects including 
+    /// extruding disjoint meshes along a vertex normal. This behavior is designed to be used in conjunction 
+    /// with the MRTK/Standard shader which assumes smoothed normals are packed into the 3rd UV set.
+    /// </summary>
+    [RequireComponent(typeof(MeshFilter))]
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_MRTKStandardShader.html#mesh-outlines")]
+    public class MeshSmoother : MonoBehaviour
+    {
+        private const int smoothNormalUVChannel = 2;
```

Comment:
This seems more like an editor level utility? What is the benefit of running this at runtime as a MonoBehavior?

Reply:
We need to do this at runtime for meshes that are loaded dynamically. Such as loading user generated content at runtime.